app/Http/Controllers/Admin/CategoryController.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Traits\Common;
+use App\Models\Category;
+use App\Models\JobData;
+
+class CategoryController extends Controller
+{
+    use Common;
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        $categories = Category::get();
+        return view('admin.categories',compact('categories'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     */
+    public function create()
+    {
+
+        return view('admin.add_category');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'category_name'=> 'required|string',
+            'image' =>'required|mimes:png,jpg,jpeg|max:2048',
+  
+        ]);
+       
+        if($request->hasFile('image')){
+            $data['image'] = $this->uploadFile($request->image,'assets/img');
+        }
+       Category::create($data);
+       return redirect()->route('categories.index');
+    }
+
+    /**
+     * Display the specified resource.
+     */
+    public function show(string $id)
```

Comment:
you can remove this if you are not planning to make this page

Reply:
Check the last edited branch